### PR TITLE
Firestore functions to interact with the events collection

### PR DESCRIPTION
**Pull request by aniruddhavpatil**
Added functions that populate, order and fetch documents in the firestore collection `events`.

1. `createEventsList()` populates the firestore database with all events in `events.json`. Resolves to `true` if operation completes, else resolves to `false`;
2. `getEventById(id)` fetches the document with id `id`. Resolves to the document fetched, else resolves to `null`.
3. `getOrderedEventsList(orderBy)` takes an `orderBy` object to fetch and sort documents by the order specified. Resolves to a list of documents fetched, else resolves to `null`.
```
let orderBy = {
            sortBy: 'Event End Date'/'Call For Proposals End Date',
            sortDirection: 'asc'/'desc',
            filters: [
                'ShowPastEvents',
                'RecentlyEndedEvents',
                'CallForProposalsOpen',
            ],
        };
```
To get the documents fetched by these, we use promise resolution:
`createEventsList().then(res => console.log(res));`
`getEventById(id).then(doc => console.log(doc));`
`getOrderedEventsList(orderBy).then(docs => console.log(docs));`

### CODE REVIEW

File: src/Pages/Home/Home.js [hunk 1]
```diff
@@ -6,7 +6,26 @@ import './Home.css';
 
 class Home extends Component {
     componentDidMount() {
-        FirestoreService.createGroceryList('Name', 'id');
+        // FirestoreService.createGroceryList('Name', 'id');
```
Comment (PrabhanshuAttri) [inline src/Pages/Home/Home.js:9]: Avoid commenting code. Either delete it or use it.

File: src/Services/firebase.js [hunk 2]
```diff
@@ -1,6 +1,8 @@
+
 import * as firebase from "firebase/app";
 import "firebase/firestore";
 import "firebase/auth";
+import * as R from "ramda";
```
Comment (PrabhanshuAttri) [inline src/Services/firebase.js:5]: Import only what you need, not all.

File: src/Services/firebase.js [hunk 3]
```diff
@@ -10,7 +12,7 @@ const config = {
     storageBucket: process.env.REACT_APP_STORAGE_BUCKET,
     messagingSenderId: process.env.REACT_APP_MESSAGING_SENDER_ID,
 };
-console.log(config)
+console.log(config);
```
Comment (PrabhanshuAttri) [inline src/Services/firebase.js:15]: delete this.

File: src/Services/firebase.js [hunk 4]
```diff
@@ -19,66 +21,77 @@ export const authenticateAnonymously = () => {
...
-                    .doc(groceryListId)
-                    .collection('items')
-                    .add({
-                        name: item,
-                        created: firebase.firestore.FieldValue.serverTimestamp(),
-                        createdBy: userId
-                    });
+export const getOrderedEventsList = orderBy => {
+    // let orderBy = {
+    //     sortBy: 'Event End Date',
```
Comment (PrabhanshuAttri) [inline src/Services/firebase.js:61]: Same as above

File: src/Services/firebase.js [hunk 5]
```diff
@@ -19,66 +21,77 @@ export const authenticateAnonymously = () => {
...
+    return new Promise((resolve, reject) => {
+        let query = db.collection(process.env.REACT_APP_COLLECTION_KEY);
+        if(!orderBy.filters.includes('ShowPastEvents')){
+            let curr_date = new Date();
+            query = query.where('Event End Date', '>=', curr_date.toISOString());
+        }
+        if(orderBy.filters.includes('RecentlyEndedEvents')){
+            let curr_date = new Date();
+            query = query.where('Event End Date', '<', curr_date.toISOString());
+            curr_date.setMonth(curr_date.getMonth() - 1);
```
Comment (PrabhanshuAttri) [inline src/Services/firebase.js:78]: Use camelCasing

File: src/Services/firebase.js [hunk 6]
```diff
@@ -19,66 +21,77 @@ export const authenticateAnonymously = () => {
     return firebase.auth().signInAnonymously();
```
Comment (PrabhanshuAttri) [inline src/Services/firebase.js:21]: Delete this function

File: src/Services/firebase.js [hunk 7]
```diff
@@ -19,66 +21,77 @@ export const authenticateAnonymously = () => {
...
+    //     filters: [
+    //         'ShowPastEvents',
+    //         'RecentlyEndedEvents',
+    //         'CallForProposalsOpen',
+    //     ],
+    // };
+    return new Promise((resolve, reject) => {
+        let query = db.collection(process.env.REACT_APP_COLLECTION_KEY);
+        if(!orderBy.filters.includes('ShowPastEvents')){
+            let curr_date = new Date();
```
Comment (PrabhanshuAttri) [inline src/Services/firebase.js:72]: Use const wherever possible

File: src/Services/firebase.js [hunk 8]
```diff
@@ -19,66 +21,77 @@ export const authenticateAnonymously = () => {
...
+            query = query.where('Event End Date', '>=', curr_date.toISOString());
```
Comment (PrabhanshuAttri) [inline src/Services/firebase.js:73]: Try creating separate methods for repetitive code.

File: src/Services/firebase.js (hunk 2 above)
Comment (aniruddhavpatil) [reply src/Services/firebase.js]: Only imported the used components.

File: src/Services/firebase.js (hunk 3 above)
Comment (aniruddhavpatil) [reply src/Services/firebase.js]: That statement has been deleted.

File: src/Services/firebase.js (hunk 4 above)
Comment (aniruddhavpatil) [reply src/Services/firebase.js]: Removed.

File: src/Services/firebase.js (hunk 5 above)
Comment (aniruddhavpatil) [reply src/Services/firebase.js]: Changed every variable to use camelCasing.

File: src/Services/firebase.js (hunk 6 above)
Comment (aniruddhavpatil) [reply src/Services/firebase.js]: Deleted.

File: src/Services/firebase.js (hunk 7 above)
Comment (aniruddhavpatil) [reply src/Services/firebase.js]: This requires the use of let in order to be able to filter using multiple where clauses.

File: src/Services/firebase.js (hunk 8 above)
Comment (aniruddhavpatil) [reply src/Services/firebase.js]: Created a whereQueryConstructor to avoid repetitive code.

**Pull request merged**